src/pymmcore_plus/mda/metadata/schema.py

```diff
@@ -0,0 +1,169 @@
+from typing import Any, Dict, List, Literal, Optional, Tuple, TypedDict
+
+import useq
+from typing_extensions import NotRequired
+
+__all__ = [
+    "FrameMetaV1",
+    "SummaryMetaV1",
+    "ConfigGroup",
+    "ConfigPreset",
+    "DeviceInfo",
+    "ImageInfo",
+    "PixelSizeConfigPreset",
+    "Position",
+    "PropertyInfo",
+    "PropertyValue",
+    "SystemInfo",
+]
+
+
+class PropertyInfo(TypedDict):
+    """Information about a single device property."""
+
+    name: str
+    value: Optional[str]
+    data_type: Literal["undefined", "float", "int", "str"]
+    allowed_values: Optional[Tuple[str, ...]]
+    is_read_only: bool
+    is_pre_init: bool
+    limits: NotRequired[Tuple[float, float]]
+    sequenceable: bool
+    sequence_max_length: NotRequired[int]
+    # device_label: str
+
+
+class DeviceInfo(TypedDict):
+    """Information about a specific device."""
+
+    label: str
+    library: str
+    name: str
+    type: str
+    description: str
+    properties: Tuple[PropertyInfo, ...]
+    parent_label: Optional[str]  # None on hub devices
+
+    # state device only
+    labels: NotRequired[Tuple[str, ...]]
+    # focus device only
+    focus_direction: NotRequired[Literal["Unknown", "TowardSample", "AwayFromSample"]]
+    # hub device only
+    child_names: NotRequired[Tuple[str, ...]]
+    # stage or XYstage device only
+    is_stage_sequenceable: NotRequired[bool]
+    # stage device only
+    is_continuous_focus_drive: NotRequired[bool]
+    is_stage_linear_sequenceable: NotRequired[bool]
```

Comment:
This linear sequencing thing is something that I once thought ASI stages could do and (sort of) implemented, but it ended up never working the way we expected. It's so obscure and probably unused that it might make sense to ignore it. Not that it harms anything to have the field.